torch/nn/modules/transformer.py

```diff
@@ -375,7 +375,7 @@ def forward(
             why_not_sparsity_fast_path = "NestedTensor input is not supported"
         elif mask is not None:
             why_not_sparsity_fast_path = "src_key_padding_mask and mask were both supplied"
-        elif torch.is_autocast_enabled():
+        elif torch.is_autocast_enabled(src.device.type):
```

Comment:
FYI @drisspg the subtle difference here is that the old code was checking only for autocast being enabled on cuda while the updated code will check for autograd being enabled on the device of the input.

Reply:
I think that should be fine, since there aren't any mixed input devices